filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,177 @@
+		return &privateKey.PublicKey, nil
+	})
+
+	if err != nil {
+		t.Errorf("Failed to json decode: %v", err)
+		return
+	}
+
+	err = parsedToken.Claims.Valid()
+
+	if err != nil {
+		t.Errorf("Failed token: %v", err)
+		return
+	}
+
+}
+
+func TestJWTValidation(t *testing.T) {
+	cli := net.NewClient(net.Options{
+		IdleConnTimeout: 2 * time.Second,
+	})
+	defer cli.Close()
+
+	backend := httptest.NewServer(http.HandlerFunc(func(http.ResponseWriter, *http.Request) {}))
+	defer backend.Close()
+
+	authServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(489)
+			return
+		}
+
+		allKeys := map[string][]interface{}{}
+		allKeys["keys"] = append(allKeys["keys"], map[string]interface{}{"kid": kid, "n": base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes()), "e": "AQAB"})
+		if r.URL.Path != testAuthPath {
+			w.WriteHeader(488)
+			return
+		}
+
+		e := json.NewEncoder(w)
+		err2 := e.Encode(allKeys)
+		if err2 != nil && err2 != io.EOF {
+			t.Errorf("Failed to json encode: %v", err2)
+		}
+	}))
+	defer authServer.Close()
+
+	testOidcConfig := getTestOidcConfig()
+	issuerServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != TokenIntrospectionConfigPath {
+			w.WriteHeader(486)
+			return
+		}
+		e := json.NewEncoder(w)
+		err := e.Encode(testOidcConfig)
+		if err != nil {
+			t.Fatalf("Could not encode testOidcConfig: %v", err)
+		}
+	}))
+	defer issuerServer.Close()
+
+	// patch openIDConfig to the current testservers
+	testOidcConfig.Issuer = "http://" + issuerServer.Listener.Addr().String()
+	testOidcConfig.JwksURI = "http://" + authServer.Listener.Addr().String() + testAuthPath
+
+	var spec = NewJwtValidation(testAuthTimeout)
+
+	args := []interface{}{testOidcConfig.Issuer}
+	f, err := spec.CreateFilter(args)
+	if err != nil {
+		t.Errorf("error in creating filter: %v", err)
+		return
+	}
+
+	f2 := f.(*jwtValidationFilter)
+	defer f2.Close()
+
+	fr := make(filters.Registry)
+	fr.Register(spec)
+	r := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: args}}, Backend: backend.URL}
+
+	proxy := proxytest.New(fr, r)
+	defer proxy.Close()
+
+	reqURL, err := url.Parse(proxy.URL)
+	if err != nil {
+		t.Errorf("Failed to parse url %s: %v", proxy.URL, err)
+		return
+	}
+
+	req, err := http.NewRequest("GET", reqURL.String(), nil)
+	if err != nil {
+		t.Errorf("failed to create request %v", err)
+		return
+	}
+	for _, ti := range []struct {
+		msg      string
+		auth     string
+		expected int
+	}{{
+		msg:      "jwtValidation: invalid token",
+		auth:     "invalid-token",
+		expected: http.StatusUnauthorized,
+	}, {
+		msg:      "jwtValidation: valid token",
+		auth:     createToken(t),
+		expected: http.StatusOK,
+	}} {
+		t.Run(ti.msg, func(t *testing.T) {
+
+			req.Header.Set(authHeaderName, authHeaderPrefix+ti.auth)
```

Comment:
Lets create request a new here
```go
req, _ := http.NewRequest("GET", proxy.URL, nil)
```

Lets move prefix to the test case field such that we can have tests like:
```go
        {
		msg:      "invalid token type",
		auth:     "Wearer " +createToken(t) ,
		expected: http.StatusUnauthorized,
	},{
		msg:      "empty toke",
		auth:     "Bearer " ,
		expected: http.StatusUnauthorized,
	},
```

Reply:
Lets also add a test for a token with `none` algorithm. The token should be valid according to JWT but the filter should refuse to accept it. See https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/